src/main/java/io/github/apickledwalrus/skriptgui/gui/GUI.java

```diff
@@ -179,7 +179,8 @@ public Character convert(Object slot) {
 
 		if (slot instanceof Number) {
 			int invSlot = ((Number) slot).intValue();
-			if (rawShape != null && invSlot < rawShape.length()) {
+			// Due to client bug slot may be -1 if not properly detected.
+			if (rawShape != null && invSlot != -1 && invSlot < rawShape.length()) {

```

Comment:
it might be better to check that slot is greater than or equal to 0 just as a precaution

Reply:
Got it. Already changed, but i think it may be more precise to only check `invSlot` wheather equals `-1`.